### PR TITLE
remove deprecated code

### PR DESCRIPTION
**Pull request by garbados**
## Proposed Release Notes

- Fix deprecated code, such as `fs.existsSync` and using `NULL` vs. the more modern `nullptr`.

## Links

- Closes https://github.com/newrelic/node-native-metrics/issues/118

### CODE REVIEW

File: lib/pre-build.js [hunk 1]
```diff
@@ -95,7 +95,7 @@ function findNodeGyp() {
   if (process.env.npm_config_node_gyp) {
     try {
       gypPath = process.env.npm_config_node_gyp
-      if (fs.existsSync(gypPath)) {
+      if (fs.accessSync(gypPath)) {
```
Comment (bizob2828) [inline lib/pre-build.js:98]: this ticket was super vague but it was suggested to just remove this entire try/catch.  But I would just leave untouched.  This was a straight ripoff of this logic https://github.com/mapbox/node-pre-gyp/blob/e0b3b6/lib/util/compile.js#L18-L55. I don't know enough of the user case where a user would set `npm_config_node_gyp` but this supports it 🤷🏻‍♂️

**Review by bizob2828 — CHANGES_REQUESTED** [review]

File: lib/pre-build.js (hunk 1 above)
Comment (garbados) [reply lib/pre-build.js:98]: Can you clarify? Should I remove the entire try/catch or just revert my changes? `existsSync` is deprecated and has been superseded by `accessSync` so if we are going to keep this block around, we might as well use modern methods, right?

Comment (bizob2828) [reply lib/pre-build.js:98]: true, that's fair

**Review by bizob2828 — DISMISSED** [review]

**Comment by garbados** [comment]
@bizob2828 Made a minor change since `accessSync` and `existsSync` have a slightly different return signature: `accessSync` will error if it encounters problems and return `undefined` otherwise, whereas `existsSync` returns a boolean. So, it is enough to simply run `accessSync` and defer to the try/catch. Putting it in an if/else would cause the if-clause to never run, because even when the file does exist `accessSync` simply returns `undefined`.

**Review by bizob2828 — DISMISSED** [review]
verified this works as expected. also provided a good and path path for `npm_config_node_gyp` and it builds successfully.  One thing to note is this feels a little weird now because the rest of the method is still using `fs.existsSync`

**Comment by garbados** [comment]
> feels a little weird

I agree, the inconsistency is off-putting. I pushed another commit that replaces all instances of `existsSync` appropriately. Sorry to keep asking you to review this! 😓

**Review by bizob2828 — APPROVED** [review]
👋🏻 existsSync, LGTM

**Pull request merged**